### PR TITLE
Deck SelectionDialog now uses Alert Dialog

### PR DESCRIPTION
**Pull request by neeldoshii**
## Fixes
* Related #13315  

## Approach
Replaces usages of Deprecated MaterialDialog and adds the Native AlertDialog For DeckSelectionDialog.

## Images

| Material Dialog | Alert Dialog |
|-----------------|--------------|
| Material Dialog | Alert Dialog |

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [X] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt [hunk 1]
```diff
@@ -64,13 +66,12 @@ import kotlin.collections.ArrayList
...
 @NeedsTest("simulate 'don't keep activities'")
 open class DeckSelectionDialog : AnalyticsDialogFragment() {
-    private var dialog: MaterialDialog? = null
+    private var dialog: AlertDialog? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         isCancelable = true
     }
 
-    @Suppress("Deprecation") // Material dialog neutral button deprecation
```
Comment (neeldoshii) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt:73]: #11780 Removed the suppress annotation as we have migrated on Alert Dialog.

**Review by BrayanDSO — COMMENTED** [review]
Works well. Thanks. I'm going to leave a patch with some improvements.

```patch
Subject: [PATCH] refactor: remove DeckSelectionDialog::dialog
---
Index: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt b/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt
--- a/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt	(revision cd0c65a055e80b5cb819d60a9944edd950e8fd5d)
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt	(revision d527d2fc4193ac9da00b195e3f495445f9c490eb)
@@ -66,7 +66,6 @@
  */
 @NeedsTest("simulate 'don't keep activities'")
 open class DeckSelectionDialog : AnalyticsDialogFragment() {
-    private var dialog: AlertDialog? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         isCancelable = true
@@ -98,7 +97,8 @@
             val did = args.getLong("currentDeckId")
             recyclerView.scrollToPosition(getPositionOfDeck(did, adapter.getCurrentlyDisplayedDecks()))
         }
-        dialog = AlertDialog.Builder(requireActivity()).apply {
+
+        return AlertDialog.Builder(requireActivity()).apply {
             negativeButton(R.string.dialog_cancel)
             customView(view = dialogView)
             if (arguments.getBoolean(KEEP_RESTORE_DEFAULT_BUTTON)) {
@@ -107,7 +107,6 @@
                 }
             }
         }.create()
-        return dialog as AlertDialog
     }
 
     private fun getPositionOfDeck(did: DeckId, decks: List<SelectableDeck>) =
@@ -221,11 +220,11 @@
      */
     protected fun selectDeckAndClose(deck: SelectableDeck) {
         onDeckSelected(deck)
-        dialog!!.dismiss()
+        dialog?.dismiss()
     }
 
     protected fun displayErrorAndCancel() {
-        dialog!!.dismiss()
+        dialog?.dismiss()
     }
 
     open inner class DecksArrayAdapter(deckNames: List<SelectableDeck>) : RecyclerView.Adapter<DecksArrayAdapter.ViewHolder>(), Filterable {
```

Consider this as an approval after the patch is applied

**Review by david-allison — COMMENTED** [review]
@BrayanDSO is there a reason you chose `dialog.dismiss()` over `dismiss()`?

----

I'll get a patch in for `create {` since it comes up a lot

----

Treat this as a soft approve once changes are made, looks good!

**Comment by BrayanDSO** [comment]
> [BrayanDSO](https://github.com/BrayanDSO) is there a reason you chose dialog.dismiss() over dismiss()?

I wrote that by reflex. Didnt thought much about It. Thanks. Edited

**Comment by neeldoshii** [comment]
I applied the patch given by @BrayanDSO waiting for `.create()` patch.

**Comment by david-allison** [comment]
> I applied the patch given by @BrayanDSO waiting for `.create()` patch.

It's merged

**Review by BrayanDSO — APPROVED** [review]

**Review by lukstbit — APPROVED** [review]
LGTM, thanks!
Left one small issue and then we could merge this.

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt [hunk 2]
```diff
@@ -97,15 +97,16 @@ open class DeckSelectionDialog : AnalyticsDialogFragment() {
...
             recyclerView.scrollToPosition(getPositionOfDeck(did, adapter.getCurrentlyDisplayedDecks()))
         }
-        dialog = MaterialDialog(requireActivity())
-            .negativeButton(R.string.dialog_cancel)
-            .customView(view = dialogView, noVerticalPadding = true)
-        if (arguments.getBoolean(KEEP_RESTORE_DEFAULT_BUTTON)) {
-            (dialog as MaterialDialog).positiveButton(R.string.restore_default) {
-                onDeckSelected(null)
+
+        return AlertDialog.Builder(requireActivity()).apply {
```
[the remark this answers is not in the document]
Comment (neeldoshii) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckSelectionDialog.kt]: Added in the current commit.

**Review by lukstbit — APPROVED** [review]

**Comment by mikehardy** [comment]
@neeldoshii 

> [Merge branch 'ankidroid:main' into deckselection-to-alert-dialog](https://github.com/ankidroid/Anki-Android/pull/15431/commits/ea45e3095206b9509f3df2317ee8e10f0686b888)

We use a rebase workflow here whenever possible. If you'll be contributing a lot, you should investigate how to use git rebase to pull new changes into a PR branch vs adding merge commits

**Comment by neeldoshii** [comment]
> @neeldoshii
> 
> > [Merge branch 'ankidroid:main' into deckselection-to-alert-dialog](https://github.com/ankidroid/Anki-Android/pull/15431/commits/ea45e3095206b9509f3df2317ee8e10f0686b888)
> 
> We use a rebase workflow here whenever possible. If you'll be contributing a lot, you should investigate how to use git rebase to pull new changes into a PR branch vs adding merge commits

Noted.

**Pull request merged**